specs/BrowserProcessExited.md

```diff
@@ -0,0 +1,322 @@
+        }
+
+        return S_OK;
+    }).Get(),
+    &browserExitedEventToken));
+```
+
+## .NET C#
+```c#
+// URI or other state to save/restore when the WebView is recreated.
+private Uri _uriToRestore;
+
+async void RegisterForNewVersion()
+{
+    // We need to make sure the CoreWebView2 property is not null, so we can get
+    // the environment from it. Alternatively, if the WebView was created from
+    // an environment provided to the control, we can use that environment
+    // object directly.
+    await webView.EnsureCoreWebView2Async();
+    _coreWebView2Environment = webView.CoreWebView2.Environment;
+    _coreWebView2Environment.NewBrowserVersionAvailable += Environment_NewBrowserVersionAvailable;
+}
+
+// A new version of the WebView2 Runtime is available, our handler gets called.
+// We close our WebView and set a handler to reinitialize it once the browser
+// process is gone, so we get the new version of the WebView2 Runtime.
+void Environment_NewBrowserVersionAvailable(object sender, object e)
+{
+    StringBuilder messageBuilder = new StringBuilder(256);
+    messageBuilder.Append("We detected there is a new version of the WebView2 Runtime installed. ");
+    messageBuilder.Append("Do you want to switch to it now? This will re-create the WebView.");
+    var selection = MessageBox.Show(this, messageBuilder.ToString(), "New WebView2 Runtime detected", MessageBoxButton.YesNo);
+    if (selection == MessageBoxResult.Yes)
+    {
+        // Save URI or other state you want to restore when the WebView is recreated.
+        _uriToRestore = webView.Source;
+        _coreWebView2Environment.BrowserProcessExited += Environment_BrowserProcessExited;
+        // We dispose of the control so the internal WebView objects are released
+        // and the associated browser process exits. If there are any other WebViews
+        // from the same environment configuration, they need to be closed too.
+        webView.Dispose();
+        webView = null;
+    }
+}
+
+void Environment_BrowserProcessExited(object sender, CoreWebView2BrowserProcessExitedEventArgs e)
+{
+    ((CoreWebView2Environment)sender).BrowserProcessExited -= Environment_BrowserProcessExited;
+    ReinitializeWebView();
+}
+
+void ReinitializeWebView()
+{
+    webView = new WebView2();
+
+    // Restore URI and other WebView state/setup.
+    webView.CreationProperties = (CoreWebView2CreationProperties)this.FindResource("EvergreenWebView2CreationProperties");
+    webView.NavigationStarting += WebView_NavigationStarting;
+    webView.NavigationCompleted += WebView_NavigationCompleted;
+
+    Binding urlBinding = new Binding()
+    {
+        Source = webView,
+        Path = new PropertyPath("Source"),
+        Mode = BindingMode.OneWay
+    };
+    url.SetBinding(TextBox.TextProperty, urlBinding);
+
+    MyWindow.MyDockPanel.Children.Add(webView);
+    webView.Source = (_uriToRestore != null) ? _uriToRestore : new Uri("https://www.bing.com");
```

Comment:
Can say `_uriToRestore ?? new Url("https://www.bing.com")` to save typing.

Reply:
Updated.